infra/discovery/src/listener/listener.js

```diff
@@ -114,9 +114,13 @@ async function main() {
   // List of contracts the listener watches events from.
   const contracts = {
     IdentityEvents: contractsContext.identityEvents,
-    Marketplace: contractsContext.marketplaces['000'].contract,
     ProxyFactory: contractsContext.ProxyFactory
   }
+  // Listen to all versions of marketplace
+  Object.keys(contractsContext.marketplaces).forEach(key => {
+    contracts[`V${key}_Marketplace`] =
```

Comment:
One thing I just realized was `key` is 3 characters long (eg '000') but the config if 2 (eg 'V00'). Not sure if that affects this. Also might need to say `Object.keys(contractsContext.marketplaces || {})` incase no marketplaces are set yet for whatever reason

Reply:
it doesn't have any effect here, it's using the keys from the same object to reference it.  But maybe that should be standardized across the codebase.